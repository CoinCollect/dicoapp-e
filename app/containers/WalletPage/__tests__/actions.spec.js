import {
  loadTransactions,
  loadTransactionsSuccess,
  openWithdrawModal,
  closeWithdrawModal
} from '../actions';
import {
  LOAD_TRANSACTIONS,
  LOAD_TRANSACTIONS_SUCCESS,
  WITHDRAW_MODAL_OPEN,
  WITHDRAW_MODAL_CLOSE
} from '../constants';

describe('containers/WalletPage/actions/loadTransactions', () => {
  it('should loadTransactions should create loadTransactions action', () => {
    expect(loadTransactions()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_TRANSACTIONS
    };

    expect(loadTransactions()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadTransactionsSuccess', () => {
  it('should loadTransactionsSuccess should create loadTransactionsSuccess action', () => {
    expect(loadTransactionsSuccess()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_TRANSACTIONS_SUCCESS
    };

    expect(loadTransactionsSuccess()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/openWithdrawModal', () => {
  it('should loadTransactionsSuccess should create openWithdrawModal action', () => {
    expect(openWithdrawModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_MODAL_OPEN
    };

    expect(openWithdrawModal()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/closeWithdrawModal', () => {
  it('should loadTransactionsSuccess should create closeWithdrawModal action', () => {
    expect(closeWithdrawModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: WITHDRAW_MODAL_CLOSE
    };

    expect(closeWithdrawModal()).toEqual(expectedResult);
  });
});
