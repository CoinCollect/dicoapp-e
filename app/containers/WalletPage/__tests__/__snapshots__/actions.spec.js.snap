// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/WalletPage/actions/closeWithdrawModal should loadTransactionsSuccess should create closeWithdrawModal action 1`] = `
Object {
  "type": "dicoapp/WalletPage/WITHDRAW_MODAL_CLOSE",
}
`;

exports[`containers/WalletPage/actions/loadTransactions should loadTransactions should create loadTransactions action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS",
}
`;

exports[`containers/WalletPage/actions/loadTransactionsSuccess should loadTransactionsSuccess should create loadTransactionsSuccess action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS_SUCCESS",
}
`;

exports[`containers/WalletPage/actions/openWithdrawModal should loadTransactionsSuccess should create openWithdrawModal action 1`] = `
Object {
  "type": "dicoapp/WalletPage/WITHDRAW_MODAL_OPEN",
}
`;
